java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -765,21 +770,16 @@ private void buildTryStatement(TryStatementTree tryStatementTree) {
     }
     currentBlock = beforeFinally;
     build(tryStatementTree.block());
-    if(currentBlock.exitBlock != null && currentBlock.exitBlock.isFinallyBlock) {
-      for (Block catchBlock : catches) {
-        currentBlock.exitBlock.addSuccessor(catchBlock);
-      }
-    } else {
-      for (Block catchBlock : catches) {
-        currentBlock.addSuccessor(catchBlock);
-      }
+    if (currentBlock != beforeFinally) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)